src/core/main.cpp

```diff
@@ -567,6 +567,72 @@ int main( int argc, char * * argv )
 		}
 	}
 
+	// Test file argument before continuing
+
+	QFile testFile( fileToLoad );
+	if( testFile.open( QIODevice::ReadOnly) )
+	{
+		// Test if render ( coreOnly ) and file is empty.
+		if( coreOnly == true && testFile.size() == 0 )
+		{
+			// is empty
+			printf( "The file %s is empty.\n",
+					fileToLoad.toStdString().c_str() );
+			exit( 1 );
+		}
+		testFile.close();
+	}
+
+	if( !fileToLoad.isEmpty() )
+	{
+		QFileInfo fileInfoLoad( fileToLoad );
+		if( !fileInfoLoad.exists() )
```

Comment:
`!fileInfoLoad.size()` checks if the file does not exist or its size is zero. The error message could be `The file %s does not have any content`.